test/tests/functional/pbs_job_array.py

```diff
@@ -394,8 +394,8 @@ def test_job_array_history_duration(self):
         b = {'job_state': 'X'}
         self.server.expect(JOB, b, subjid_1)
         self.server.expect(JOB, a, subjid_2, attrop=PTL_AND)
-        self.server.expect(JOB, 'job_state', op=UNSET, id=subjid_1, interval=5)
-        self.server.expect(JOB, 'job_state', op=UNSET, id=subjid_2, interval=5)
+        self.server.expect(JOB, 'job_state', op=UNSET, id=subjid_1, offset=30)
+        self.server.expect(JOB, 'job_state', op=UNSET, id=subjid_2, offset=30)
```

Comment:
Why offset=30?  The jobs only sleep 5s and you delete subjob 1.  Also, don't you need an extend='t' to do an expect on a subjob?

Reply:
Offset is for finish of history duration , I need to add extend='x'

Reply:
It does a qstat -f for subjob id , So i dont think extend='t' is necessary

Reply:
OK, I read through the code.  For the most part the server purges jobs out of the history every 2m (it can be less under certain circumstances).  I think this means the worst case scenario will be a delay of 2:30 until the job is deleted.  You could make this a little shorter if you shrunk the history duration some.  I'd worry that if you made it too short, you could run into race conditions.